### PR TITLE
Add handling to uppercase value

### PR DESCRIPTION
**Pull request by PashkovichKaterina**
**Task:** Handle shouting. When name is all uppercase, then the method should shout back to the user. For example, when name is "JERRY" then the method should return the string "HELLO JERRY!" Write a test for this method using Pytest.

**Solution steps:**

1. Creating test for uppercase value
2. Add handling to uppercase value in function

### CODE REVIEW

File: greeting_test.py [hunk 1]
```diff
@@ -7,3 +7,7 @@ def test_greet():
 
 def test_greet_null():
     assert "Hello, my friend." in greet(None)
+
+
+def test_greet_shouting():
+    assert "HELLO JERRY!" in greet("JERRY")
```
Comment (artsiom-streetshares) [inline greeting_test.py:13]: the test will work for these cases:
"HELLO HERRY!sfsdfs"
"sfsdfsHELLO HERRY!"
"HELLO HERRY!HELLO HERRY!"

File: greeting.py [hunk 2]
```diff
@@ -1,4 +1,8 @@
 def greet(name):
     if name is None:
         name = "my friend"
-    return "Hello, {}.".format(name)
+    if isinstance(name, str) and name.isupper():
```
Comment (artsiom-streetshares) [inline greeting.py:4]: are you expecting something other than str or None? if no, better check at the beginning that type is correct and further on assume it's str. this way it's more readable.

File: greeting_test.py (hunk 1 above)
Comment (PashkovichKaterina) [reply greeting_test.py]: Thank you. Test is fixed.

File: greeting.py (hunk 2 above)
Comment (PashkovichKaterina) [reply greeting.py]: Thank you. Did changes in the function.

**Review by artsiom-streetshares — APPROVED** [review]

**Pull request merged**